File: C#/LeetCode/20. Valid Parentheses.cs
public class Solution {
    public bool IsValid(string s) 
    {
        Dictionary<char, char> dictionary = new Dictionary<char, char>();
            Stack<char> stack = new Stack<char>();

            dictionary.Add(')', '(');
            dictionary.Add('}', '{');
            dictionary.Add(']', '[');

            foreach (var str in s)
            {
                if (str is ')' or ']' or '{')
                {
                    if (stack.Count > 0 && stack.Peek() == dictionary[str])
                    {
                        stack.Pop();
                    }
                    else
                    {
                        return false;
                    }
                }
                else
                {
                    stack.Push(str);
                }
            }

            return stack.Count == 0;
    }
}

File: LeetCode/Python/20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        bracketMap = {")" : "(", "]": "[", "}": "{"}
        
        for l in s:
            if l in bracketMap:
                if stack and stack[-1] == bracketMap[l]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(l)
        if not stack:
            return True
        else:
            return False
        
